main/sceneBuilder.h

```diff
@@ -0,0 +1,68 @@
+#ifndef SCENEBUILDER_H
+#define SCENEBUILDER_H
+
+#include "constants.h"
+#include "core/vec3.h"
+#include "core/ray.h"
+#include "obj/generators.h"
+#include "obj/objects.h"
+
+#include <memory>
+#include <string>
+#include <string_view>
+#include <vector>
+
+
+class SceneBuilder
+{
+    public:
+    SceneBuilder() : _collectorCurrentId(0) {};
+
+    // Construct Sceneshould move ownership of the objects to the class Simulator
```

Comment:
Some questions:

- what is "construct", given that this is called BuildEnergyCollectors()?
- what is Simulator?
- if you want to return energy collectors, why do you return pointers to objects::Object? Just return pointers to energy collectors 

- "should use private method build collector for situating it at the right position" do not know what that means. it seems to me like this comment describes how it should do it (so implementation) not what it should do. I would remove it, as for users of this class it does not matter if this will call a private methods or will do it in a different way, as long as all collectors are in the right place.

Reply:
good point 👍. construct is garbage word. I forgot to delete it probably. Scene is another class object, that I plan to implement. It will be clear in the future. 

""should use private method build collector for situating it at the right position" do not know what that means. it seems to me like this comment describes how it should do it (so implementation) not what it should do. I would remove it, as for users of this class it does not matter if this will call a private methods or will do it in a different way, as long as all collectors are in the right place."
This wont be in final version of the file ;)

Reply:
Please read carefully the bottom of the comment about the clean interface. This is quite important to get right.

Reply:
noted. I added interface to my comments and prepared it for later discussion.